crates/accelerate/src/high_level_synthesis.rs

```diff
@@ -171,82 +213,807 @@ impl QubitTracker {
+/// applicable (for instance, when there is an insufficient number of auxiliary qubits).
+///
+/// Internally, this function calls the Python function.
+///
+/// Currently, this function does not update the qubit tracker, which is handled upstream.
+fn synthesize_op_using_plugins(
+    py: Python,
+    data: &Bound<HighLevelSynthesisData>,
+    tracker: &Bound<QubitTracker>,
+    input_qubits: &[usize],
+    op: &OperationRef,
+    params: &[Param],
+    label: Option<&str>,
+) -> PyResult<Option<(CircuitData, Vec<usize>)>> {
+    let mut output_circuit_and_qubits: Option<(CircuitData, Vec<usize>)> = None;
+
+    let op_py = match op {
+        OperationRef::StandardGate(standard) => {
+            standard.create_py_op(py, Some(params), label)?.into_any()
+        }
+        OperationRef::StandardInstruction(instruction) => instruction
+            .create_py_op(py, Some(params), label)?
+            .into_any(),
+        OperationRef::Gate(gate) => gate.gate.clone_ref(py),
+        OperationRef::Instruction(instruction) => instruction.instruction.clone_ref(py),
+        OperationRef::Operation(operation) => operation.operation.clone_ref(py),
+        OperationRef::Unitary(unitary) => unitary.create_py_op(py, label)?.into_any(),
+    };
+
+    let res = HLS_SYNTHESIZE_OP_USING_PLUGINS
+        .get_bound(py)
+        .call1((op_py, input_qubits, data, tracker))?
+        .extract::<Option<(QuantumCircuitData, Vec<usize>)>>()?;
+
+    if let Some((quantum_circuit_data, qubits)) = res {
+        output_circuit_and_qubits = Some((quantum_circuit_data.data, qubits));
+    }
+
+    Ok(output_circuit_and_qubits)
+}
+
+/// Synthesizes an operation.
+///
+/// This function is currently called by the default plugin for annotated operations to
+/// synthesize the base operation.
+#[pyfunction]
+#[pyo3(signature = (py_op, input_qubits, data, tracker))]
+fn py_synthesize_operation(
+    py: Python,
+    py_op: Bound<PyAny>,
+    input_qubits: Vec<usize>,
+    data: &Bound<HighLevelSynthesisData>,
+    tracker: &Bound<QubitTracker>,
+) -> PyResult<Option<(CircuitData, Vec<usize>)>> {
+    let op: OperationFromPython = py_op.extract()?;
+
+    // Check if the operation can be skipped.
+    if definitely_skip_op(
+        py,
+        data,
+        &op.operation,
+        &input_qubits
+            .iter()
+            .map(|q| Qubit(*q as u32))
+            .collect::<Vec<Qubit>>(),
+    ) {
+        return Ok(None);
+    }
+
+    synthesize_operation(
+        py,
+        data,
+        tracker,
+        &input_qubits,
+        &op.operation,
+        &op.params,
+        op.label.as_ref().map(|x| x.as_str()),
+    )
+}
+
+/// Runs HighLevelSynthesis transpiler pass.
+///
+/// This is the main function called from the Python space. If the pass does not need
+/// to do anything, it returns None, meaning that the DAG should remain unchanged.
+/// Otherwise, the new DAG is returned.
+#[pyfunction]
+#[pyo3(signature = (dag, data, qubits_initially_zero))]
```

Comment:
```suggestion
#[pyo3(name = "run_on_dag", signature = (dag, data, qubits_initially_zero))]
```

As with `py_synthesize_operation`.